lib/src/themes/expression/parser/expression_parser.dart

```diff
@@ -96,6 +96,13 @@ class ExpressionParser {
       }
     }
     if (json is Map) {
+      if (json['type'] == 'categorical') {
+        final property = json['property'];
+        final stops = json['stops'];
+        final defaultValue = json['default'];
+        return CategoricalPropertyExpression(property, defaultValue, stops);
```

Comment:
Looks like we could have some cast exceptions here if property and defaultValue are not strings, or if stops is not a list. Can we have more robust type safety? See below for examples. Maybe we could have some code reuse too - e.g. the check for stops below.